### DISCUSSION TITLE
Using /extras/config-templates/<id>/render/

### DISCUSSION THREAD

**Discussion by ed-ud** [netbox-community/netbox#16798] [Q&A]
From the example here:

https://netboxlabs.com/docs/netbox/en/stable/features/configuration-rendering/#general-purpose-use

... in the template, how do I access "foo" and "bar"?

With a template like:

Foo is {{ foo }}
Bar is {{ bar }}

... the values are empty.

-- Ed

**Comment by ed-ud** [comment]
Does anyone have a clue how to access the "context data" for this endpoint?

**Comment by candlerb** [comment]
Netbox version? Did you remember to include data in the POST to the API endpoint?

It works for me on 4.0.11 following the [example](https://netboxlabs.com/docs/netbox/en/stable/features/configuration-rendering/#general-purpose-use).

```
# curl -X POST \
-H "Authorization: Token $TOKEN" \
-H "Content-Type: application/json" \
-H "Accept: application/json; indent=4" \
"$NETBOX/api/extras/config-templates/4/render/" \
--data '{
  "foo": "abc",
  "bar": 123
}'
```

Response:

```
{
    "configtemplate": {
        "id": 4,
        "url": "https://netbox.example.net/api/extras/config-templates/4/",
        "display": "Foo bar",
        "name": "Foo bar",
        "description": ""
    },
    "content": "Foo is abc\nBar is 123"
}
```

... and if you change the Accept header to `Accept: text/plain` then you get the raw result:

```
Foo is abc
Bar is 123
```

Content of template:

```
Foo is {{ foo }}
Bar is {{ bar }}
```

**Comment by ed-ud** [comment]
Thanks Brian!  Oddly enough that worked for me too.  My python program was failing to do the very same thing... but it turned out to be a "programmer error" and the data that was supposedly getting passed to the template was getting replaced before the actual requests.post() call ... DOH!
